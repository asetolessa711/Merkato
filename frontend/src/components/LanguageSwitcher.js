
import React from 'react';

const LANGUAGES = [
  { code: 'en', label: 'English' },
  { code: 'am', label: 'Amharic' },
  { code: 'om', label: 'Oromiffa' },
  { code: 'sw', label: 'Swahili' },
];

const LanguageSwitcher = ({ currentLang, setLang }) => (
  <div>
    {LANGUAGES.map(lang => (
      <button
        key={lang.code}
        className={currentLang === lang.code ? 'active' : ''}
        onClick={() => {
          if (setLang && lang.code !== currentLang) setLang(lang.code);
        }}
        aria-label={lang.label}
      >
        {lang.label}
      </button>
    ))}
  </div>
);

export default LanguageSwitcher;
