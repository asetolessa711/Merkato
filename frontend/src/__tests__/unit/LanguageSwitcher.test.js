import React from 'react';
import { render, screen, fireEvent } from '@testing-library/react';
import LanguageSwitcher from '../../components/LanguageSwitcher';
import '@testing-library/jest-dom';

const languages = [
  { code: 'en', label: /english/i },
  { code: 'am', label: /amharic/i },
  { code: 'om', label: /oromiffa/i },
  { code: 'sw', label: /swahili/i }
];

describe('\ud83c\udf0d Language Switcher', () => {
  test('renders all available language options', () => {
    render(<LanguageSwitcher currentLang="en" />);
    languages.forEach(lang =>
      expect(screen.getByText(lang.label)).toBeInTheDocument()
    );
  });

  test('triggers language change on selection', () => {
    const mockSetLang = jest.fn();
    render(<LanguageSwitcher currentLang="en" setLang={mockSetLang} />);
    fireEvent.click(screen.getByText(/amharic/i));
    expect(mockSetLang).toHaveBeenCalledWith('am');
  });

  test('does not trigger callback when clicking current language', () => {
    const mockSetLang = jest.fn();
    render(<LanguageSwitcher currentLang="en" setLang={mockSetLang} />);
    fireEvent.click(screen.getByText(/english/i));
    expect(mockSetLang).not.toHaveBeenCalled();
  });

  test('highlights selected language', () => {
    render(<LanguageSwitcher currentLang="sw" />);
    const active = screen.getByText(/swahili/i);
    expect(active).toHaveClass('active'); // Adjust class name per implementation
  });

  test('all language options are accessible as buttons', () => {
    render(<LanguageSwitcher currentLang="en" />);
    languages.forEach(lang => {
      expect(screen.getByRole('button', { name: lang.label })).toBeInTheDocument();
    });
  });

  test('matches snapshot', () => {
    const { asFragment } = render(<LanguageSwitcher currentLang="en" />);
    expect(asFragment()).toMatchSnapshot();
  });
});
