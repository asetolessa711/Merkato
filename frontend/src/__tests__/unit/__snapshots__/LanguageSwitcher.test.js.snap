// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`🌍 Language Switcher matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <button
      aria-label="English"
      class="active"
    >
      English
    </button>
    <button
      aria-label="Amharic"
      class=""
    >
      Amharic
    </button>
    <button
      aria-label="Oromiffa"
      class=""
    >
      Oromiffa
    </button>
    <button
      aria-label="Swahili"
      class=""
    >
      Swahili
    </button>
  </div>
</DocumentFragment>
`;
